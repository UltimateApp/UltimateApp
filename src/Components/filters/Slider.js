import React from 'react';
import { StyleSheet, View, Text } from 'react-native';
import { Slider as NativeSlider } from 'react-native-elements';
import theme from '../../styles/theme.style';

const Slider = ({ minimumValue, maximumValue, step, value, onValueChange, testID }) => {
  return (
    <View style={styles.container}>
      <Text style={styles.boundaries}>{minimumValue}</Text>
      <NativeSlider
        minimumValue={minimumValue}
        maximumValue={maximumValue}
        step={step}
        value={value}
        onValueChange={onValueChange}
        style={styles.slider}
        trackStyle={styles.trackStyle}
        minimumTrackTintColor={theme.FORM_ELEMENT_COLOR}
        thumbTintColor={theme.FORM_ELEMENT_COLOR}
        testID={testID}
      />
      <Text style={styles.boundaries}>{maximumValue}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 10,
    flexDirection: 'row',
    alignItems: 'center',
  },
  boundaries: {
    textAlign: 'center',
    flexGrow: 1,
  },
  slider: {
    flexGrow: 10,
    marginHorizontal: 10,
  },
  trackStyle: {
    height: 2,
  },
});

export default Slider;
