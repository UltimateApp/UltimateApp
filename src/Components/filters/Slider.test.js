import React from 'react';
import renderer from 'react-test-renderer';

import Slider from './Slider';

describe('<Slider />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Slider value={15} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
