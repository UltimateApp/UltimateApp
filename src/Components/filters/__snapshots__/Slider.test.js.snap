// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "paddingHorizontal": 10,
    }
  }
>
  <Text
    style={
      Object {
        "flexGrow": 1,
        "textAlign": "center",
      }
    }
  />
  <View
    animationType="timing"
    onLayout={[Function]}
    step={0}
    style={
      Object {
        "flexGrow": 10,
        "height": 40,
        "justifyContent": "center",
        "marginHorizontal": 10,
      }
    }
    theme={
      Object {
        "colors": Object {
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "primary": "#2196f3",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "ios": Object {
              "error": "#ff3b30",
              "primary": "#007aff",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#8F0CE8",
          "success": "#52c41a",
          "warning": "#faad14",
        },
      }
    }
    thumbTouchSize={
      Object {
        "height": 40,
        "width": 40,
      }
    }
    value={15}
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "backgroundColor": "#b3b3b3",
          "borderRadius": 2,
          "height": 2,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#08AEEA",
          "borderRadius": 2,
          "height": 0,
          "marginTop": -0,
          "position": "absolute",
          "width": 0,
        }
      }
    />
    <View
      onLayout={[Function]}
      style={
        Object {
          "backgroundColor": "#08AEEA",
          "borderRadius": 10,
          "height": 0,
          "position": "absolute",
          "top": 21,
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": -0,
            },
          ],
          "width": 0,
        }
      }
      testID="sliderThumb"
    />
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "flexGrow": 1,
        "textAlign": "center",
      }
    }
  />
</View>
`;
